app/controllers/katello/concerns/api/v2/hosts_controller_extensions.rb

```diff
@@ -20,6 +20,14 @@ def host_collections
           @host.save!
           render(:locals => { :resource => @host }, :template => 'katello/api/v2/hosts/show', :status => 200)
         end
+
+        alias_method :foreman_index, :index
+        def index
+          foreman_index
+          @hosts = @hosts.includes([
+            :content_view, :lifecycle_environment, :subscription_facet, :content_host, :applicable_errata
+          ]).includes(content_facet: :bound_repositories)
+        end
```

Comment:
Foreman has a host facet API that we plugin to to extend the Host object, here's our use of it:

https://github.com/Katello/katello/blob/master/lib/katello/engine.rb#L228-L236

and here's the documentation:
http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Facets